server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/DatabaseChangeThread.java

```diff
@@ -245,6 +213,59 @@ public void run() {
         }
     }
 
+    private void retryStatement(RetryCallBack<List<JdbcGeneralResult>> retryCallBack, JdbcOperations executor,
+            StatementCallback<List<JdbcGeneralResult>> statementCallback, String sql) {
+        boolean success = true;
+        GeneralSqlType sqlType = parseSqlType(sql);
+        for (int retryCount = 0; retryCount <= parameters.getRetryTimes(); retryCount++) {
+            success = true;
+            try {
+                queryResultSetBuffer.clear();
+                List<JdbcGeneralResult> results =
+                        executor.execute((StatementCallback<List<JdbcGeneralResult>>) stmt -> {
+                            stmt.setQueryTimeout(
+                                    (int) TimeUnit.MILLISECONDS.toSeconds(parameters.getTimeoutMillis()));
+                            return statementCallback.doInStatement(stmt);
+                        });
+                Verify.notEmpty(results, "resultList");
+                for (JdbcGeneralResult result : results) {
+                    SqlExecuteResult executeResult = new SqlExecuteResult(result);
+                    if (GeneralSqlType.DQL == sqlType) {
+                        this.isContainQuery = true;
+                        if (maskingService.isMaskingEnabled()) {
+                            try {
+                                dynamicDataMasking(executeResult);
+                            } catch (Exception e) {
+                                // Eat exception and skip data masking
+                                log.warn("Failed to mask query result set in database change task, sql={}",
+                                        executeResult.getExecuteSql(), e);
+                            }
+                        }
+                    }
+                    queryResultSetBuffer.add(executeResult);
+
+                    if (executeResult.getStatus() != SqlExecuteStatus.SUCCESS) {
+                        log.warn("Error occurs when executing sql: {}, error message: {}", sql,
+                                executeResult.getTrack());
+                        success = false;
+                        retryCallBack.track = executeResult.getTrack();
+                    }
+                }
+            } catch (Exception e) {
+                if (retryCount == parameters.getRetryTimes()) {
```

Comment:
When `retryCount` < `retryTimes` but error occurs when `executor.execute()`, then the `success` is still `true` and the retry will be break.  Anyway, please check the code logic.

Reply:
Sure. Very careful!